mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowProtobufMapper.java

```diff
@@ -78,6 +81,49 @@ String makeLogBatch(String runUuid,
     return print(builder);
   }
 
+
+  public static Metric createMetric(String name, double value, long timestamp) {
+    Metric.Builder builder = Metric.newBuilder();
+    builder.setKey(name).setValue(value).setTimestamp(timestamp);
+    return builder.build();
+  }
+
+  List<Metric> makeMetricList(Map<String, Double> metrics) {
+    List<Metric> metricList = new ArrayList<>();
+    for (Map.Entry<String, Double> entry: metrics.entrySet()) {
+      metricList.add(createMetric(entry.getKey(), entry.getValue(), System.currentTimeMillis()));
```

Comment:
One small comment, maybe compute the timestamp ahead of this for loop so all the metrics are logged with the same timestamp (otherwise it could differ across metrics e.g. for a large list)

Reply:
So something like:

```
    long timestamp = System.currentTimeMillis();
    for (Map.Entry<String, Double> entry: metrics.entrySet()) {
      metricList.add(createMetric(entry.getKey(), entry.getValue(), timestamp));
    }

```

Reply:
Ha!I was trying to avoid duplicate metrics -- but there wouldn't be due to the interface and we already have a fix in place in case, it happens.

Great call!